risc0/circuit/rv32im/src/prove/emu/exec/mod.rs

```diff
@@ -437,6 +437,15 @@ impl<'a, 'b, S: Syscall> Executor<'a, 'b, S> {
         if into_guest_len > 0 && !is_guest_memory(into_guest_ptr.0) {
             bail!("{into_guest_ptr:?} is an invalid guest address");
         }
+
+        if into_guest_len > 0 && !into_guest_ptr.is_null() {
+            let end_addr = into_guest_ptr
+                .0
+                .checked_add(into_guest_len as u32)
+                .ok_or_else(|| anyhow!("invalid guest address range"))?;
```

Comment:
Maybe we could add a `checked_add` to `ByteAddr` and `WordAddr`

Reply:
i'll address this in the afternoon